docs/log-levels.md

```diff
@@ -1,6 +1,6 @@
 # Log Levels
 
-This guide serves as guidelines for library authors with regards to what [SwiftLog](https://github.com/apple/swift-log) log levels are appropriate for use in libraries, and in what situations to use what level.
+This guide serves as guidelines for library authors with regards to what [SwiftLog](https://github.com/apple/swift-log) log levels are appropriate for use in libraries, and in which situations to use which level.
```

Comment:
Good introduction but could use some fleshing out.  Define what a library is here, and why logging guidelines that are suitable for applications may not be suitable for libraries.  You address specific points in subsections but could use a thesis here.

Reply:
thanks, I revamped it a bit:

```
# Libraries: Log Level guidelines for Library authors

This guide serves as guidelines for library authors with regard to what [SwiftLog](https://github.com/apple/swift-log) log levels are appropriate for use in libraries, and in what situations to use what level.

Libraries need to be well-behaved across various use cases, and cannot assume a specific style of logging backend will be used with them. It is up to developers implementing specific applications and systems to configure those specifics of their application, and some may choose to log to disk, some to memory, or some may employ sophisticated log aggregators. In all those cases a library should behave "well", meaning that it should not overwhelm typical ("stdout") log backends by logging too much, alerting too much by over-using `error` level log statements etc.

In this guide we'll explore the various aspects a library developer should be aware of when implementing logging in their library.

```